src/fb5.js

```diff
@@ -234,18 +239,46 @@ function setupPageObserver() {
 
 // cleanup
 window.addEventListener("beforeunload", () => {
-  pageObserver.disconnect();
+  cleanupPageObserver();
 
   if (feedObserver !== null) {
-    feedObserver.disconnect();
-    feedObserver = null;
+    cleanupFeedObserver();
   }
 
   if (watchObserver !== null) {
-    watchObserver.disconnect();
-    watchObserver = null;
+    cleanupWatchObserver();
   }
 });
+function cleanupPageObserver() {
+  if (pageObserver === null) return;
+  pageObserver.__observed &&
+    delete feedObserver.__observed.dataset.adblockObserved &&
+    delete feedObserver.__observed;
+  pageObserver.disconnect();
+  // do not nullify pageObserver!
```

Comment:
Can you clarify why we cannot do this?

Reply:
Actually we could clear the pageObserver too at this point since this cleanupPageObserver is only called when the beforeunload event triggers, meaning that page will change and if facebook is reloaded, the script will rerun and initialization will occur [on line 205](https://github.com/siradar/facebook_adblock/blob/master/src/fb5.js#L205). The only reason not to nullify it here is because we don't have a mechanism to initiate it again, like we do with the rest observers, without re-running the script.